node/pkg/governor/governor_prices.go

```diff
@@ -128,14 +128,35 @@ func (gov *ChainGovernor) PriceQuery(ctx context.Context) error {
 // return an error, but that is only used by the tool that validates the query. In the actual governor,
 // it just logs the error and we will try again next interval. If an error happens, any tokens that have
 // not been updated will be assigned their pre-configured price.
-func (gov *ChainGovernor) queryCoinGecko() error {
+func (gov *ChainGovernor) queryCoinGecko(ctx context.Context) error {
 	result := make(map[string]interface{})
 
 	// Cache buster of Unix timestamp concatenated with random number
 	params := url.Values{}
 	params.Add("bust", strconv.Itoa(int(time.Now().Unix()))+strconv.Itoa(rand.Int())) // #nosec G404
 
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	// Throttle the queries to the CoinGecko API. We query for 200 tokens at a time, so this throttling would
+	// allow us to query up to 12,000 tokens in a 15 minute window (the query interval). Currently there are
+	// between 1000 and 2000 tokens.
+	throttle := make(chan int, 1)
```

Comment:
As a followup, for your governor update script, I almost wonder if it makes sense to add a check for `len(governed_tokens) > 12000`? If that fails, there will be something in the test to tell you to come update this. Not for this PR, but makes sense as a followup I think to keep everyone honest.

Reply:
Great suggestion! Have made a note to open a PR for this